build.rs

```diff
@@ -0,0 +1,65 @@
+use std::{env, fs::File, io::prelude::*, path::PathBuf};
+
+fn main() {
+    if cfg!(feature = "ld") {
+        gen_memory_x();
+    }
+    println!("cargo:rerun-if-changed=build.rs");
+}
+
+fn gen_memory_x() {
+    #![allow(clippy::unneeded_wildcard_pattern)]
+
+    let mem_cfg_set = (
```

Comment:
Hm. Isn't an enum a better option here? Then the following code should be much more readable.